backend/src/main/java/com/summerHack/diningTogether/model/UserDetails.java

```diff
@@ -13,9 +13,10 @@ public class UserDetails implements org.springframework.security.core.userdetail
     String password;
 
     Long id;
-
+    User user;
+    //might be redundant here
```

Comment:
It is redundant.

If you want to disable the user if it is not verified, you can use a field called `enabled`, and only store that field from user object in the `of` method below.